### PR TITLE
[FTR] Split environment configs to separate files

### PR DESCRIPTION
**Pull request by Den4ik**
Related Issue: #98

### CODE REVIEW

File: environments/magento2.db.base.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - --explicit_defaults_for_timestamp=on
+    volumes:
+      - dbdata:/var/lib/mysql
+    networks:
+      - warden
+      - default
+
+volumes:
+  dbdata:
+  esdata:
```
[the remark this answers is not in the document]
Comment (Den4ik) [reply environments/magento2.db.base.yml:31]: Thanks @ihor-sviziev. This PR should be reworked. I prepared another configs for internal usage and update this PR soon

File: commands/env.cmd [hunk 2]
```diff
@@ -18,6 +18,42 @@ DOCKER_COMPOSE_ARGS=()
 DOCKER_COMPOSE_ARGS+=("-f")
 DOCKER_COMPOSE_ARGS+=("${WARDEN_DIR}/environments/${WARDEN_ENV_TYPE}.base.yml")
```
Comment (ihor-sviziev) [inline commands/env.cmd:20]: I think would be good to name file as `mariadb`, not as `db`, so there will be ability to use MySQL instead

**Comment by davidalger** [comment]
After a long wait, I've integrated this into a feature branch that accomplishes this and more: `feature/environment-template-restructure` and plan to have that branch supersede this PR

That branch builds off of the (already shuffled environment yml structuring on current develop) to de-duplicate much of the service configuration allowing them to be included piece by piece, and then merged in the branch from this PR (resolving all the conflicts) to introduce the feature flags for use in project .env files. When integrating this PR into the new feature branch, the flags have been given default values so that if they remain unspecified the services will function as they did before. I.e. not require the project .env file be updated to explicitly specify default services for each environment preventing a BC break.

What @ihor-sviziev mentioned about the MARIADB flag being too specific is accurate and I plan to change the variable name from WARDEN_MARIADB to WARDEN_DB before merging. A subsequent PR can add variables to specify an alternate image to switch from MariaDB to MySQL as desired, but I do plan to keep the files named as simply `db` since it matches the service name in the YAML (and thus hostname used in existing project database connection configuration).

**For my own future reference.**

For testing the branch to ensure the overhual doesn't significantly alter the content of the env configs themselves when assembled, I have a test directory setup with a handful of env configurations:

```
$ tree -a
.
├── laravel
│   └── .env
├── local
│   └── .env
├── magento1
│   └── .env
├── magento1-allon
│   └── .env
├── magento2
│   └── .env
└── magento2-allon
    └── .env

6 directories, 6 files
```

From within this directory, I then run the following to diff the configuration:

```
for OSTYPE in darwin19 linux-gnu; do export OSTYPE; for ENV_TYPE in $(ls -1 .); do echo "==> comparing ${ENV_TYPE}_${OSTYPE}"; (cd ${ENV_TYPE}; diff -u <(/opt/warden-0.4/bin/warden env config) <(/opt/warden/bin/warden env config)) done done
```

**Comment by Den4ik** [comment]
Thanks @davidalger It's really good job. I'm close this PR

**Pull request closed without merging**